### PR TITLE
Feature/bootstrap key view

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Description

This is a first step to help the compiler integration effort on the cuda front where they will need to get the pointer of the container for a standard bootstrap key, the workflow for the bootstrap key views for the ffi still needs some discussion.

We may need a discarding creation for lwe bootstrap keys or engines to allow to copy to a view

### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        (InputKeyDistribution, OutputKeyDistribution),
+        Engine,
+        (Container, BootstrapKey),
+    > for LweBootstrapKeyConstructionFixture
+where
+    Precision: IntegerPrecision,
+    InputKeyDistribution: KeyDistributionMarker,
+    OutputKeyDistribution: KeyDistributionMarker,
+    Engine: LweBootstrapKeyConstructionEngine<Container, BootstrapKey>,
+    BootstrapKey: LweBootstrapKeyEntity, // Todo, do someting for the key distribution
```
Comment (agnesLeroy) [inline concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs:35]: Why is there this todo here?

**Review by agnesLeroy — APPROVED** [review]
Looks good to me! :) Just a small question about a todo in a fixture. Thanks for doing this so quickly!

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs (hunk 1 above)
Comment (IceTDrinker) [reply concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs:35]: it was before the keydistribution marker changes and good thing I put it there I forgot it x)

**Comment by IceTDrinker** [comment]
I think I forgot some generic size check, I'll add it at the same time as the fixture key distribution todo

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs (hunk 1 above)
Comment (agnesLeroy) [reply concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs:35]: :smile:

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_consuming_retrieval.rs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    >
+    Fixture<
+        Precision,
+        (InputKeyDistribution, OutputKeyDistribution),
+        Engine,
+        (LweBootstrapKey, Container),
+    > for LweBootstrapKeyConsumingRetrievalFixture
+where
+    Precision: IntegerPrecision,
+    Precision: IntegerPrecision,
```
Comment (aPere3) [inline concrete-core-fixture/src/fixture/lwe_bootstrap_key_consuming_retrieval.rs:41]: Oups !

**Review by aPere3 — APPROVED** [review]
Hello @IceTDrinker ! 

Thanks for this PR ! It's cool to see more views coming to the default backend :)

The PR looks good to me, you just duplicated a bound in a fixture, but otherwise, it looks good to me !

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_consuming_retrieval.rs (hunk 2 above)
Comment (IceTDrinker) [reply concrete-core-fixture/src/fixture/lwe_bootstrap_key_consuming_retrieval.rs:41]: thanks resolved locally

File: concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs (hunk 1 above)
Comment (IceTDrinker) [reply concrete-core-fixture/src/fixture/lwe_bootstrap_key_construction.rs:35]: done locally, I'll update the size checks for the container

**Comment by IceTDrinker** [comment]
@slab-ci cpu_test

**Pull request merged**